instrumentation/net_ldap/lib/opentelemetry/instrumentation/net/ldap/patches/instrumentation.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Net
+      module LDAP
+        module Patches
+          # Module to prepend to Net::LDAP for instrumentation
+          module Instrumentation
+            def initialize(args = {})
+              super
+
+              @instrumentation_service = args[:instrumentation_service] || OpenTelemetry::Instrumentation::Net::LDAP::InstrumentationService.new({
```

Comment:
Another thought I have here, what if we added this otel instrumentation as a decorator to the injected service?